isis/src/mro/apps/hideal2pds/main.cpp

```diff
@@ -114,14 +117,47 @@ void IsisMain() {
   pair<double, double> inRange;
   inRange = inputRange(inputCube);
   p.SetInputRange(minmin, maxmax);
-  // output bit type will be 16bit unsigned word
-  p.SetOutputType(Isis::UnsignedWord); 
-  p.SetOutputNull(Isis::NULLU2);         
-  p.SetOutputLrs(Isis::LOW_REPR_SATU2);  
-  p.SetOutputLis(Isis::LOW_INSTR_SATU2); 
-  p.SetOutputHrs(Isis::HIGH_REPR_SATU2); 
-  p.SetOutputHis(Isis::HIGH_INSTR_SATU2);
-  p.SetOutputRange(VALID_MINU2, VALID_MAXU2); 
+
+   int nbits = ui.GetInteger("BITS");
+
+   switch (nbits) {
+
+
+    case 8:
+     g_oType = Isis::UnsignedWord;
+     p.SetOutputType(g_oType);
+     p.SetOutputRange(VALID_MIN1, VALID_MAX1);
+     p.SetOutputNull(NULL1);
+     p.SetOutputLis(LOW_INSTR_SAT1);
+     p.SetOutputLrs(LOW_REPR_SAT1);
+     p.SetOutputHis(HIGH_INSTR_SAT1);
+     p.SetOutputHrs(HIGH_REPR_SAT1);
+     break;
+
+    case 16:
+     g_oType = UnsignedWord;
+     p.SetOutputType(g_oType);
+     p.SetOutputRange(VALID_MINU2, VALID_MAXU2);
+     p.SetOutputNull(NULLU2);
+     p.SetOutputLis(LOW_INSTR_SATU2);
+     p.SetOutputLrs(LOW_REPR_SATU2);
+     p.SetOutputHis(HIGH_INSTR_SATU2);
+     p.SetOutputHrs(HIGH_REPR_SATU2);
+     break;
+
+   default:
+     g_oType = UnsignedWord;
+     p.SetOutputType(g_oType);
+     p.SetOutputRange(3.0, pow(2.0, (double)(nbits)) - 1.0 - 2.0);
+     p.SetOutputNull(0);
+     p.SetOutputLrs(1);
+     p.SetOutputLis(2);
+     p.SetOutputHis(pow(2.0, (double)(nbits)) - 1.0 - 1.0);
+     p.SetOutputHrs(pow(2.0, (double)(nbits)) - 1.0);
```

Comment:
Setting the default in this switch to 10 bit output can result in users entering something odd like 12 bit output and then getting 10 bit output. This logic should be under the 10 bit case and then the default case should raise an exception about requesting an unsupported output type.

Reply:
Not sure how the user can specify 12 bits and end up getting 10. Can you explain? The switch case default handles all bit outputs from 9 to 15. The 10 bit default is set in the xml. The xml test has a BITS min of 8 and max of 16, so no error checking is necessary here. The code would be unreachable.